lib/wasi/src/syscalls/mod.rs

```diff
@@ -959,13 +959,11 @@ pub fn fd_prestat_dir_name<M: MemorySize>(
     let guard = inode_val.read();
     match guard.deref() {
         Kind::Dir { .. } | Kind::Root { .. } => {
-            // TODO: verify this: null termination, etc
             let path_len: u64 = path_len.into();
-            if (inode_val.name.len() as u64) < path_len {
+            if (inode_val.name.len() as u64) <= path_len {
                 wasi_try_mem!(path_chars
                     .subslice(0..inode_val.name.len() as u64)
                     .write_slice(inode_val.name.as_bytes()));
-                wasi_try_mem!(path_chars.index(inode_val.name.len() as u64).write(0));
```

Comment:
We might still want to terminate it by 0 (but just not pass it to the user).
Basically, all the PR as is but, reverting this to what it was:
```
                wasi_try_mem!(path_chars.index(inode_val.name.len() as u64).write(0));
```

Thoughts?

Reply:
we might overwite memory here! The memory is allocated client side, for the result of `prestat_fd`, where the 0 is not counted anymore on this PR.
We cannot have both. It with or without the appended 0

Reply:
Ok!